File: Core/actor/actormessage.cpp
#include "actormessage.h"

ActorMessage::ActorMessage(QString actorGroupId, QString actorId, QVariant rawValue, ACTOR_CMDS cmd, QObject *parent) :
    ValueMessage(actorGroupId, actorId, rawValue), m_cmd(cmd)
{

}

MessageBase::MESSAGE_TYPE ActorMessage::getMessageType() {
    return MESSAGE_TYPE_ACTOR;
}

ACTOR_CMDS ActorMessage::cmd() {
    return m_cmd;
}

File: AndroidApp/OshCore/src/main/java/com/osh/actor/ActorMessage.java
package com.osh.actor;

import static com.osh.value.ValueBase.VALUE_SEPARATOR;

import com.osh.communication.MessageBase;
import com.osh.value.ValueMessage;

public class ActorMessage extends ValueMessage {
	private ActorCmds cmd;
	
	public ActorMessage(String actorGroupId, String actorId, Object value, ActorCmds cmd) {
		super(actorGroupId, actorId, value);
		this.cmd = cmd;
	}

	public ActorMessage(ActorBase actor, Object value, ActorCmds actorCmd) {
		this(actor.getValueGroup().getId(), actor.getId(), value, actorCmd);
	}

	@Override
	public MESSAGE_TYPE getMessageType() {
		return MESSAGE_TYPE.MESSAGE_TYPE_ACTOR;
	}


	public ActorCmds getCmd() {
		return cmd;
	}

}
